### PR TITLE
[flutter_tools] add more docs to cocoapods, move to globals

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Add some placeholder docs to the Cocoapods validator and Cocoapods class. Add the injected validator to globals.

https://github.com/flutter/flutter/issues/47161

### CODE REVIEW

File: packages/flutter_tools/lib/src/macos/cocoapods.dart [hunk 1]
```diff
@@ -66,6 +66,13 @@ enum CocoaPodsStatus {
   brokenInstall,
 }
 
+/// Cocoapods is a depenency management solution for iOS and macOS applications.
+///
+/// Cocoapods is generally installed via homebrew and interacted with via
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/macos/cocoapods.dart:71]: I believe our official recommendation is to use rubygems, e.g. https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/macos/cocoapods.dart#L44

**Review by christopherfujino — APPROVED** [review]
nit about s/homebrew/rubygems/, otherwise lgtm

File: packages/flutter_tools/lib/src/macos/cocoapods.dart (hunk 1 above)
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/macos/cocoapods.dart:71]: fyi, i think the homebrew formula for cocoapods just delegates to rubygems anyway

Comment (jmagman) [reply packages/flutter_tools/lib/src/macos/cocoapods.dart:71]: +1 I have never installed it with homebrew.
https://flutter.dev/docs/get-started/install/macos#deploy-to-ios-devices

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/macos/cocoapods.dart:71]: TIL, why did I install it with brew - perhaps that was old documentation?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/macos/cocoapods.dart]: Done, and added link to iOS getting started docs

**Review by jmagman — APPROVED** [review]
LGTM

**Pull request merged**